irc.go

```diff
@@ -54,6 +54,89 @@ func (ms *modeSet) Apply(s string) error {
 	return nil
 }
 
+type channelModeType byte
+
+const (
+	modeTypeA channelModeType = iota
+	modeTypeB
+	modeTypeC
+	modeTypeD
+)
+
+var stdChannelModes = map[byte]channelModeType{
+	'b': modeTypeA,
+	'e': modeTypeA,
+	'I': modeTypeA,
+	'k': modeTypeB,
+	'l': modeTypeC,
+	'i': modeTypeD,
+	'm': modeTypeD,
+	'n': modeTypeD,
+	's': modeTypeD,
+	't': modeTypeD,
+}
+
+func ApplyChannelModes(modeTypes map[byte]channelModeType, modes map[byte]string, modeStr string, arguments ...string) error {
```

Comment:
Just like we have `userModes`, maybe it makes sense to have `type channelModes map[byte]string` with methods like `Apply` and `String`?

Reply:
OK. I'm not sure about "`String()`" though b/c we don't return a string but also a string array.

Reply:
What about `Format` or `Marshal`?

Reply:
`Format` is good, i'll do that